src/Cake.Common.Tests/Unit/Tools/NuGet/List/NuGetListTests.cs

```diff
@@ -0,0 +1,250 @@
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.ForceEnglishOuput = true;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -ForceEnglishOuput -NonInteractive", result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_IncludeDelisted_To_Arguments_If_True()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.IncludeDelisted = true;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -IncludeDelisted -NonInteractive", result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_Prerelease_To_Arguments_If_True()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.Prerelease = true;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -Prerelease -NonInteractive", result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_Sources_To_Arguments_If_Set()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.Source = new[] { "A;B;C" };
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -Source \"A;B;C\" -NonInteractive", result.Args);
+            }
+
+            [Theory]
+            [InlineData(NuGetVerbosity.Detailed, "list \"Cake\" -Verbosity detailed -NonInteractive")]
+            [InlineData(NuGetVerbosity.Normal, "list \"Cake\" -Verbosity normal -NonInteractive")]
+            [InlineData(NuGetVerbosity.Quiet, "list \"Cake\" -Verbosity quiet -NonInteractive")]
+            public void Should_Add_Verbosity_To_Arguments_If_Set(NuGetVerbosity verbosity, string expected)
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.Verbosity = verbosity;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal(expected, result.Args);
+            }
+
+            [Fact]
+            public void Should_Add_ConfigFile_To_Arguments_If_Set()
+            {
+                // Given
+                var fixture = new NuGetListFixture();
+                fixture.Settings.ConfigFile = "./nuget.config";
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("list \"Cake\" -ConfigFile \"/Working/nuget.config\" " +
+                             "-NonInteractive", result.Args);
+            }
+        }
```

Comment:
Could we also have some test for parsing the output?

Reply:
I've added a test that would check the output of the parsing. However I'm not yet completely happy, since there is no real error checking in the current implementation either although neither is there in the other code.

Reply:
If the NuGet server returns an error (processExitCode != 0), then the script throws a Cake exception, before the parsing even begins. 
```
Executing: "C:/repo/tools/NuGet.exe" list "Cake" -Source "LocalServer" -Verbosity Normal -NonInteractive
The specified source 'LocalServer' is invalid. Please provide a valid source.
An error occurred when executing task 'Default'.
Error: System.AggregateException: One or more errors occurred. ---> Cake.Core.CakeException: NuGet: Process returned an error (exit code 1).
```
Should that exception be handled?